File: solutions/typescript/darts/1/darts.ts
const squaredRadiusToScore = [
  [10 * 10, 0],
  [5 * 5, 1],
  [1 * 1, 5],
  [-1, 10],
] as const

export function score(x: number, y: number): 0 | 1 | 5 | 10 {
  const radiusSquare = x * x + y * y;
  for (const [threshold, score] of squaredRadiusToScore) {
    if (radiusSquare > threshold) {
      return score
    }
  }
  return 0
}

File: solutions/kotlin/darts/1/src/main/kotlin/Darts.kt
import kotlin.math.hypot

object Darts {
    fun score(x: Number, y: Number): Int {
        val radius = hypot(x.toDouble(), y.toDouble())
        return when {
            radius <= 1.0 -> 10
            radius <= 5.0 -> 5
            radius <= 10.0 -> 1
            else -> 0
        }
    }
}